### PR TITLE
feat: ca specs update and registered user response added

### PR DESCRIPTION
**Pull request by deepak1725**
CA Specs in Sync with CA Response Updated.
Registered User Response Handled.

Jira: https://issues.redhat.com/browse/APPAI-1449 

Specs: https://app.swaggerhub.com/apis/deepak1725/analytics-core_api_specs_for_version_2/2.1#/CA%20batch%20call/post_component_analyses_

Signed-off-by: Deepak Sharma <deepshar@redhat.com>

### CODE REVIEW

File: api_specs/v2/component_analyses.yaml [hunk 1]
```diff
@@ -149,74 +149,163 @@ components:
...
+          recommended_version : "2"
+          message: "3 vulnerabilities within this dependency, 0 exploitable vulnerabilities"
+          highest_severity: "high"
+          known_security_vulnerability_count: 2
+          security_advisory_count: 1
+          exploitable_vulnerabilities_count: 0
+          vendor_package_link: "https://snyk.io/vuln/maven:org.zeroturnaround%3Azt-zip"
         - package: "mango"
           version: "2.1"
-          recommendation:
```
Comment (prashbnair) [inline api_specs/v2/component_analyses.yaml:205]: this field is still there in the spec definition

File: bayesian/utility/v2/ca_response_builder.py [hunk 2]
```diff
@@ -433,6 +437,56 @@ def get_cve_maps(self) -> List[Dict]:
...
+    def get_premium_response(self) -> Dict:
+        """Get Premium Response.
+
+        :return: Dict of Registered User Response.
+        """
+        exploitable_vuls = self.get_exploitable_cves_counter()
+        logger.debug("Generating Premium Response.")
+        response = dict(
+            package=self.package,
+            version=self.version,
```
Comment (prashbnair) [inline bayesian/utility/v2/ca_response_builder.py:449]: The names of the fields in the premium response are different from the normal response, will this work in the UI?

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: api_specs/v2/component_analyses.yaml [hunk 3]
```diff
@@ -149,74 +149,163 @@ components:
...
-                is_private: false
-            recommended_version : "2"
-            message: "3 vulnerabilities within this dependency, 0 exploitable vulnerabilities"
-            highest_severity: "high"
-            public_vulnerabilities_count: 2
-            private_vulnerabilities_count: 1
-            exploitable_vulnerabilities_count: 0
-            vendor_package_link: "https://snyk.io/vuln/maven:org.zeroturnaround%3Azt-zip"
+          vulnerabilities:
+            - id: "SNYK-2010-3082"
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:232]: Overall there is 3 Major Difference in response:
|                                   | Free User         | Registered User |
|-----------------------------------|-------------------|-----------------|
| exploitable_vulnerabilities_count | No                | Yes             |
| premium_message                   | No                | Yes             |
| link                              | registration_link | package_link    |

Please see Specs here
https://app.swaggerhub.com/apis/deepak1725/analytics-core_api_specs_for_version_2/2.1#/CA%20batch%20call/post_component_analyses_

File: bayesian/utility/v2/ca_response_builder.py (hunk 2 above)
Comment (deepak1725) [reply bayesian/utility/v2/ca_response_builder.py:449]: @prashbnair There is only 1 field name change

In Free Response, there is `registration_link` whereas in Premium Response we will get `vendor_package_link`
and
1 field is added in Premium Response, which was not a part of FreeTier Response: `exploitable_vulnerabilities_count`

That's what we settled with, in https://app.swaggerhub.com/apis/deepak1725/analytics-core_api_specs_for_version_2/2.1#/CA%20batch%20call/post_component_analyses

Comment (prashbnair) [reply bayesian/utility/v2/ca_response_builder.py:449]: Why do you have the package, version fields?
The component_analyses field is missing, severity is named highest_severity.

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: bayesian/utility/v2/ca_response_builder.py (hunk 2 above)
Comment (deepak1725) [reply bayesian/utility/v2/ca_response_builder.py:449]: > Why do you have the package, version fields?
@prashbnair This is Requested change by you to include version, package in each Package Recommendation.
I think we had a discussion regarding this in https://github.com/fabric8-analytics/fabric8-analytics-server/pull/655

>  The component_analyses field is missing, severity is named highest_severity.
Again this discussion happened in the mentioned PR. Please note I'm not unilaterally changing anything.

File: api_specs/v2/component_analyses.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:232]: No, there is no difference in the Vulnerability level.

File: api_specs/v2/component_analyses.yaml [hunk 4]
```diff
@@ -149,74 +149,163 @@ components:
...
-              - id: "SYNK-2018-1002"
-                cvss: "1.3"
-                is_private: false
-            recommended_version : "2"
-            message: "2 vulnerabilities within this dependency"
-            highest_severity: "high"
-            public_vulnerabilities_count: 2
-            private_vulnerabilities_count: 1
-            registeration_link: "https://app.snyk.io/login"
+          vulnerabilities:
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:152]: In PR #655, There was a proposal to move schema towards more cleaner level. The proposal was to change 
`componenet_analyses->vulnerabilities->cve_info[array]`  nesting to `vulnerabilities->cve_info[array]` only.

I'm Not introducing any new changes, but all these changes are already discussed and agreed.

File: api_specs/v2/component_analyses.yaml [hunk 5]
```diff
@@ -149,74 +149,163 @@ components:
...
+                - "CVE-2010-4534"
+              fixed_in:
+                - "1.1.3"
+                - "1.2.4"
+          recommended_version : "2"
+          message: "2 vulnerabilities within this dependency"
+          highest_severity: "high"
+          known_security_vulnerability_count: 2
+          security_advisory_count: 1
+          registeration_link: "https://app.snyk.io/login"
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:186]: ack!

File: api_specs/v2/component_analyses.yaml [hunk 6]
```diff
@@ -278,21 +367,28 @@ components:
       value:
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:367]: As `exploitable_vul_count` is missing, the example is of FreeTier Response Please note with few fields modified, same can be replicated for Registered User

File: api_specs/v2/component_analyses.yaml [hunk 7]
```diff
@@ -149,74 +149,163 @@ components:
...
-                cvss: "1.3"
-                is_private: false
-            recommended_version : "2"
-            message: "3 vulnerabilities within this dependency, 0 exploitable vulnerabilities"
-            highest_severity: "high"
-            public_vulnerabilities_count: 2
-            private_vulnerabilities_count: 1
-            exploitable_vulnerabilities_count: 0
-            vendor_package_link: "https://snyk.io/vuln/maven:org.zeroturnaround%3Azt-zip"
+          vulnerabilities:
```
[the remark this answers is not in the document]
Comment (deepak1725) [reply api_specs/v2/component_analyses.yaml:231]: ack! Removed Recommendation nesting from Every Schema

**Review by prashbnair — APPROVED** [review]

**Pull request merged**